manifests/config.pp

```diff
@@ -45,4 +45,15 @@
     mode   => '0755',
   }
 
+  if $katello::enable_ostree {
+    exec { "foreman-rake-ostree-apipie":
+      command     => "/usr/sbin/foreman-rake apipie:cache:index && touch /usr/share/foreman/tmp/ostree_apipie_index_done",
+      user        => $::foreman::user,
+      environment => sort(join_keys_to_values({'HOME' => $::foreman::app_root}, '=')),
+      logoutput   => 'on_failure',
+      refreshonly => true,
+      creates => '/usr/share/foreman/tmp/ostree_apipie_index_done',
+      require  => Package[$katello::rubygem_katello_ostree],
+    }
+  }
```

Comment:
This seems like overkill. Could you now in https://github.com/Katello/puppet-katello/blob/master/manifests/install.pp#L10 add a notify to `Exec['foreman-rake-apiepie:cache:index']` ?